clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -3260,6 +3260,156 @@ test_replay_as_traced_i6107_workaround()
 #endif
 }
 
+static void
+test_replay_as_traced_dup_start()
+{
+#ifdef HAS_ZIP
+    // Test what i#6712 fixes: duplicate start entries.
+    std::cerr << "\n----------------\nTesting replay as-traced dup starts\n";
+
+    static constexpr int NUM_INPUTS = 3;
+    static constexpr int NUM_OUTPUTS = 2;
+    static constexpr int NUM_INSTRS = 6;
+    static constexpr memref_tid_t TID_A = 100;
+    static constexpr memref_tid_t TID_B = TID_A + 1;
+    static constexpr memref_tid_t TID_C = TID_A + 2;
+    static constexpr int CPU_0 = 6;
+    static constexpr int CPU_1 = 7;
+    static constexpr uint64_t TIMESTAMP_BASE = 100;
+
+    std::vector<trace_entry_t> inputs[NUM_INPUTS];
+    for (int input_idx = 0; input_idx < NUM_INPUTS; input_idx++) {
+        memref_tid_t tid = TID_A + input_idx;
+        inputs[input_idx].push_back(make_thread(tid));
+        inputs[input_idx].push_back(make_pid(1));
+        // These timestamps do not line up with the schedule file but
+        // that does not cause problems and leaving it this way
+        // simplifies the testdata construction.
+        inputs[input_idx].push_back(make_timestamp(TIMESTAMP_BASE));
+        for (int instr_idx = 0; instr_idx < NUM_INSTRS; ++instr_idx) {
+            inputs[input_idx].push_back(make_instr(42 + instr_idx));
+        }
+        inputs[input_idx].push_back(make_exit(tid));
+    }
+
+    // Synthesize a cpu-schedule file with duplicate starts.
+    std::string cpu_fname = "tmp_test_cpu_i6712.zip";
+    {
+        zipfile_ostream_t outfile(cpu_fname);
+        {
+            // Simple dup start: non-consecutive but in same output.
```

Comment:
This comment applies to L3302 and L3304 right? Please move it to L3304. I wasn't initially aware what a "dup-start" was.

Reply:
Done.